File: 3rd Sem/NM/python/exponentialregression.py
import math

# Input x and y values
x = list(map(float, input("Enter x values (space-separated): ").split()))
y = list(map(float, input("Enter y values (space-separated): ").split()))

n = len(x)

# Transform y values using natural log
log_y = [math.log(val) for val in y]

# Calculate summations
xy_sum = sum(x[i] * log_y[i] for i in range(n))
x_square_sum = sum(x[i] ** 2 for i in range(n))
x_sum = sum(x)
log_y_sum = sum(log_y)

# Calculate slope (b) and intercept (ln(a))
b1 = (n * xy_sum - x_sum * log_y_sum) / (n * x_square_sum - x_sum ** 2)
b0 = (log_y_sum - b1 * x_sum) / n
a = math.exp(b0)

# Display the exponential regression equation
print(f"Exponential Regression Equation: y = {a:.4f} * e^({b1:.4f}x)")

File: 3rd Sem/NM/C-program/exponentialregression.c
#include <stdio.h>
#include <math.h>

int main() {
    int n, i;
    printf("Enter number of data points: ");
    scanf("%d", &n);

    double x[n], y[n], log_y[n], xy_sum = 0, x_square_sum = 0, x_sum = 0, log_y_sum = 0, b0, b1;

    printf("Enter x values: ");
    for (i = 0; i < n; i++) {
        scanf("%lf", &x[i]);
    }

    printf("Enter y values: ");
    for (i = 0; i < n; i++) {
        scanf("%lf", &y[i]);
        log_y[i] = log(y[i]);
        xy_sum += x[i] * log_y[i];
        x_square_sum += x[i] * x[i];
        x_sum += x[i];
        log_y_sum += log_y[i];
    }

    b1 = (n * xy_sum - x_sum * log_y_sum) / (n * x_square_sum - x_sum * x_sum);
    b0 = (log_y_sum - b1 * x_sum) / n;

    printf("Exponential Regression Equation: y = e^(%.4f) * e^(%.4fx)\n", b0, b1);

    return 0;
}
